nevergrad/benchmark/frozenexperiments.py

```diff
@@ -4,16 +4,92 @@
 # LICENSE file in the root directory of this source tree.
 
 import typing as tp
+from uuid import uuid4 as uuid_gen
 import numpy as np
 from nevergrad import optimizers
 from nevergrad.optimization.base import ConfiguredOptimizer
 from nevergrad.optimization import experimentalvariants  # pylint: disable=unused-import
 from nevergrad.functions import ArtificialFunction
+from nevergrad.functions.perfcap3d.core import Perfcap3DExperimentConfig, Perfcap3DFunction, Perfcap3DServerComm
 from .xpbase import registry
 from .xpbase import create_seed_generator
 from .xpbase import Experiment
+
 # pylint: disable=stop-iteration-return, too-many-nested-blocks
 
+def perfcap_experiment(experiment_filename: str, seedg: tp.Iterator[tp.Optional[int]]) -> tp.Iterator[Experiment]:
+    """
+    Factory function that creates Experiments for Perfcap3DFunction
+    """
+
+    cfg = Perfcap3DExperimentConfig(experiment_filename)
+    server_comm = Perfcap3DServerComm(cfg.server_config)
+
+    budgets = [2000, 4000, 7000]
+    optimizer_names = [ "Shiwa", "RandomSearch",
+                        "RealSpacePSO", "Powell", "DiscreteOnePlusOne",
+                        "CMA", "NGO", "TBPSA", "chainCMAPowell", "DE"]
+
+    # pylint: disable=stop-iteration-return
+    for budget in budgets:
+        for optim in optimizer_names:
+            yield Experiment(function=Perfcap3DFunction(server_comm, cfg.objective_function_config, str(uuid_gen())),
+                                optimizer=optim, budget=budget, seed=next(seedg))
+
+@registry.register
+def perfcap_bench1(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
+    seedg = create_seed_generator(seed)
+    return perfcap_experiment("experiment1.json", seedg)
```

Comment:
I tend to really hate repeated code (it's basically a maintenance hell in the long run) :D
So we'd need something else for this. Since benchmarks are only accessed through the register, we can probably convert all this into a for loop that does the registration over some object. Not sure exactly how yet though so I may end up doing it later on.
Another option would be to define all the experiments in one, do you run all them independently? I feel you would usually want to try them all together. If the experiment number is provided as a descriptor of the Experiment, then the plotting tools will differenciate between them.

Reply:
I added a dynamic way to register experiments. (dynexpreg.py)